book/src/rfc/const_generic.md

```diff
@@ -0,0 +1,265 @@
+## Summary
+Enable const generic to improve modularization and reusability of the code.
```

Comment:
maybe a good idea to precise that this is const generics for functions, and not for types